### ISSUE TITLE
linter: `promise/no-multiple-resolved` false positive when `clearTimeout` and `resolve()` share a `setTimeout` callback

### ISSUE THREAD

**Issue by tusharsingh308** [oxc-project/oxc#25528]
### What version of Oxlint are you using?

1.78.0

### What command did you run?

```bash
oxlint -A all -D promise/no-multiple-resolved --promise-plugin repro.js
```

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

Minimal config (equivalent to the CLI flags above):

```jsonc
{
  "plugins": ["promise"],
  "rules": {
    "promise/no-multiple-resolved": "error"
  }
}
```

### What happened?

`promise/no-multiple-resolved` reports a false positive when a function passed to `setTimeout` contains both a timer-clear call (`clearTimeout`, `clearInterval`, or `cancelAnimationFrame`) and an explicit `resolve()` call from a Promise executor.

**Minimal reproduction:**

```javascript
const abortableDelay = (ms, signal) =>
    new Promise(resolve => {
        let timer;
        const finish = () => {
            clearTimeout(timer);
            signal?.removeEventListener('abort', finish);
            resolve();
        };
        timer = setTimeout(finish, ms);
        if (signal?.aborted) {
            finish();
            return;
        }
        signal?.addEventListener('abort', finish);
    });
```

**Actual output:**

```
repro.js:7:13: error promise(no-multiple-resolved): Promise should not be resolved multiple times. Promise is already resolved on line 5.
```

Line 5 is `clearTimeout(timer)` — oxlint treats it as the first resolve site and flags the real `resolve()` on line 7 as a duplicate.

**Expected:** No diagnostic. There is only one `resolve()` call on any execution path; `clearTimeout` does not settle the Promise.

**Additional observations:**

1. Commenting out `clearTimeout` silences the error.
2. Reordering so `resolve()` comes before `clearTimeout` moves the reported "first resolve" to the `resolve()` line and flags `clearTimeout` as the duplicate instead.
3. The false positive also triggers with only `setTimeout` (no abort listener):

   ```javascript
   new Promise(resolve => {
       let timer;
       const finish = () => {
           clearTimeout(timer);
           resolve();
       };
       timer = setTimeout(finish, ms);
   });
   ```

4. `clearInterval` and `cancelAnimationFrame` behave the same way.
5. Other statements before `resolve()` (e.g. `console.log`, `removeEventListener`) do **not** trigger the error.
6. Extracting cleanup into a separate helper avoids the false positive:

   ```javascript
   const clearTimer = () => clearTimeout(timer);
   const finish = () => {
       clearTimer();
       resolve();
   };
   ```

This pattern is common for abortable delays (timer + `AbortSignal`). The runtime behavior is correct; only the static analysis is wrong.

**Rule source:** [`no_multiple_resolved.rs`](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/promise/no_multiple_resolved.rs)
